pkg/lockfile/dpkg-status.go

```diff
@@ -142,11 +146,10 @@ func ParseDpkgStatus(pathToLockfile string) ([]PackageDetails, error) {
 		}
 
 		if pkg.Name == "" {
-			_, _ = fmt.Fprintf(
-				os.Stderr,
-				"warning: malformed DPKG status file. Found no package name in record. File: %s\n",
+			diag.Warn(fmt.Sprintf(
+				"warning: malformed DPKG status file. Found no package name in record. File: %s",
```

Comment:
🤔 I don't think we _need_ to the `warning:` prefix. 

The type information alludes to that being `lockfile.Diagnostics.Warnings` at the moment. This seems like a display thing that an application would _add_ to an unstructured output like STDERR.

As a library consumer, I would know it is a diagnostic warning, and I can choose the display/styling in a CLI or whatever. I would much rather want to see a warning string like:

```console
found malformed DPKG status file %q
```

Reply:
Totally agree, I just didn't change it because this way the output with this change is 1:1 with what it is currently - I know this sort of CLI output shouldn't be considered breaking to change, but I think it's nice to start with being as close to 1:1 as possible as a way of proving its stability.